src/storage/mvcc/txn.rs

```diff
@@ -165,17 +165,19 @@ impl<'a> MvccTxn<'a> {
                 (lock.lock_type, lock.short_value.take())
             }
             _ => {
-                return match try!(self.reader.get_txn_commit_ts(key, self.start_ts)) {
-                    // Committed by concurrent transaction.
-                    Some(_) => Ok(()),
-                    // Rollbacked by concurrent transaction.
+                return match try!(self.reader.get_txn_commit_info(key, self.start_ts)) {
+                    Some((_, WriteType::Rollback)) |
                     None => {
+                        // TODO:None should not appear
+                        // Rollbacked by concurrent transaction.
                         info!("txn conflict (lock not found), key:{}, start_ts:{}, commit_ts:{}",
                               key,
                               self.start_ts,
                               commit_ts);
                         Err(Error::TxnLockNotFound)
                     }
+                    // Committed by concurrent transaction.
+                    _ => Ok(()),
```

Comment:
I think `Some((_, WriteType::Put)) | Some((_, WriteType::Del)) => ` is more clear.